cg/meta/backup/backup.py

```diff
@@ -84,41 +82,45 @@ def fetch_flow_cell(self, flow_cell_obj: Optional[models.Flowcell] = None) -> Op
             LOG.info("%s: retrieving from PDC", flow_cell_obj.name)
 
         try:
-            pdc_flow_cell_query: list = self.query_pdc_for_flow_cell(flow_cell_obj.name)
-        except subprocess.CalledProcessError as error:
-            LOG.error("PDC query failed: %s", error.stderr)
+            pdc_flow_cell_query: List[str] = self.query_pdc_for_flow_cell(flow_cell_obj.name)
+
+        except PdcNoFilesMatchingSearchError as error:
+            LOG.error(f"PDC query failed: {error}")
             raise error
 
         archived_key: Path = self.get_archived_encryption_key_path(query=pdc_flow_cell_query)
         archived_flow_cell: Path = self.get_archived_flow_cell_path(query=pdc_flow_cell_query)
 
         if not self.dry_run:
-            start_time = get_start_time()
-            run_dir: Path = Path(self.root_dir[flow_cell_obj.sequencer_type])
-            self.retrieve_archived_key(archived_key, flow_cell_obj, run_dir)
-            self.retrieve_archived_flow_cell(archived_flow_cell, flow_cell_obj, run_dir)
+            return self._process_flow_cell(flow_cell_obj, archived_key, archived_flow_cell)
 
-            try:
-                (
-                    decrypted_flow_cell,
-                    encryption_key,
-                    retrieved_flow_cell,
-                    retrieved_key,
-                ) = self.decrypt_flow_cell(archived_flow_cell, archived_key, run_dir)
-
-                self.extract_flow_cell(decrypted_flow_cell, run_dir)
-                self.create_rta_complete(decrypted_flow_cell, run_dir)
-                self.unlink_files(
-                    decrypted_flow_cell, encryption_key, retrieved_flow_cell, retrieved_key
-                )
-            except subprocess.CalledProcessError as error:
-                LOG.error("Decryption failed: %s", error.stderr)
-                if not self.dry_run:
-                    flow_cell_obj.status = FlowCellStatus.REQUESTED
-                    self.status.commit()
-                raise error
+    def _process_flow_cell(self, flow_cell_obj, archived_key, archived_flow_cell):
+        start_time = get_start_time()
+        run_dir: Path = Path(self.root_dir[flow_cell_obj.sequencer_type])
+        self.retrieve_archived_key(archived_key, flow_cell_obj, run_dir)
+        self.retrieve_archived_flow_cell(archived_flow_cell, flow_cell_obj, run_dir)
 
-            return get_elapsed_time(start_time=start_time)
+        try:
+            (
+                decrypted_flow_cell,
+                encryption_key,
+                retrieved_flow_cell,
+                retrieved_key,
+            ) = self.decrypt_flow_cell(archived_flow_cell, archived_key, run_dir)
```

Comment:
Would be nice to type hint this tuple

Reply:
corrected for the function, dont think I can type hint in this line 👍